perfkitbenchmarker/pkb.py

```diff
@@ -376,12 +382,20 @@ def DoRunPhase(benchmark, name, spec, collector, timer):
   """
   deadline = time.time() + FLAGS.run_stage_time
   run_number = 0
+  consecutive_failures = 0
   while True:
+    samples = []
     logging.info('Running benchmark %s', name)
     events.before_phase.send(events.RUN_PHASE, benchmark_spec=spec)
     try:
       with timer.Measure('Benchmark Run'):
         samples = benchmark.Run(spec)
+    except Exception:
+      consecutive_failures += 1
+      if consecutive_failures > FLAGS.run_stage_retries:
+        raise
+    else:
+      consecutive_failures = 0
```

Comment:
The else matches the except (not the if). Sorry - hadn't pushed latest changes - the latest logs exceptions on every retry